test/groovy/MTABuildTest.groovy

```diff
@@ -228,26 +190,24 @@ public class MTABuildTest extends BasePipelineTest {
 
         new File("${currentDir}/mta.yaml") << defaultMtaYaml()
 
-        noBuildTargetPipeline()
-        def script = loadScript(pipeline)
-        script.execute()
+        withPipeline(noBuildTargetPipeline()).execute()
     }
 
 
     private defaultPipeline(){
-        new File(pipeline) <<   '''
+        { ->  '''
```

Comment:
The closure is not really needed. You do not have to write the return statement if you do not like it.

Reply:
With the current approach the closure is returned from the method call and the closure is called later on in order to return the pipeline. That is IMO a handy and common pattern.

But I also thought about inlining the pipeline string completly into the test methods.

@Reviewers: Feedback welcome about the way we should go.

Reply:
I know how the closure works, but in this case it is not needed.

Reply:
@alejandraferreirovidal it does not work for me without the closure inside the closure. Can you give us an example? Do you mean by inlining the String as @marcusholl suggests?
I favor the inlining of the pipeline scripts, even though it is repetitive code, as the developer can then see more easily what script is being executed for a specific test.

Reply:
You just have to delete the closure, and in the `PiperTestBase` in the method `withPipeline(p)` you just have to change `pipeline << p()` to `pipeline << p`